File: Python/src/array/max_consecutive_ones.py
from typing import List


class MaxConsecutiveOnes:

    def find_max_consecutive_ones(self, nums: List[int]) -> int:
        result = counter = 0
        for num in nums:
            if num == 0:
                result = max(result, counter)
                counter = 0
            else:
                counter += 1
        return max(result, counter)

File: Cpp/src/array/max_consecutive_ones.c
#include <stdio.h>

int find_max_consecutive_ones(int *nums, int numsSize) {
    int result = 0;
    int counter = 0;
    for (int i = 0; i < numsSize; i++) {
        if (nums[i] != 0) {
            counter++;
        } else {
            result = result > counter ? result : counter;
            counter = 0;
        }
    }
    result = result > counter ? result : counter;

    return result;
}
